nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-json-record-utils/src/main/java/org/apache/nifi/json/JsonParserFactory.java

```diff
@@ -23,27 +23,20 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Objects;
 
-public class JsonParserFactory implements TokenParserFactory{
-    private static final JsonFactory JSON_FACTORY = new JsonFactory();
-    private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
-
+public class JsonParserFactory implements TokenParserFactory {
     @Override
-    public JsonParser getJsonParser(InputStream in) throws IOException {
-        JsonParser jsonParser = JSON_FACTORY.createParser(in);
-        jsonParser.setCodec(JSON_MAPPER);
-
-        return jsonParser;
-    }
+    public JsonParser getJsonParser(final InputStream in, final StreamReadConstraints streamReadConstraints, final boolean allowComments) throws IOException {
+        Objects.requireNonNull(in, "Input Stream required");
+        Objects.requireNonNull(streamReadConstraints, "Stream Read Constraints required");
 
-    @Override
-    public ObjectMapper createCodec(boolean allowComments, StreamReadConstraints streamReadConstraints) {
-        ObjectMapper codec = new ObjectMapper();
-        if(allowComments) {
-            codec.enable(JsonParser.Feature.ALLOW_COMMENTS);
+        final ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.getFactory().setStreamReadConstraints(streamReadConstraints);
+        if (allowComments) {
+            objectMapper.enable(JsonParser.Feature.ALLOW_COMMENTS);
         }
-        codec.getFactory().setStreamReadConstraints(streamReadConstraints);
-
-        return codec;
+        final JsonFactory jsonFactory = objectMapper.getFactory();
+        return jsonFactory.createParser(in);
```

Comment:
The previous code in `AbstractJsonRowRecordReader.java` formally on line 146 when creating the jsonParser also called `setCodec`.  Shouldn't you have something similar here?
Instead of
` return jsonFactory.createParser(in);`

change to
```
final JsonParser jsonParser = jsonFactory.createParser(in);
jsonParser.setCodec(objectMapper);
return jsonParser;
```

Reply:
Setting the codec is not necessary because the `JsonFactory` is coming from the `ObjectMapper`, as opposed to the other way around.

Reply:
In the previous code also the `JsonFactory` was coming from the `ObjectMapper`and yet `setCodec` was still called. See the previous line 145:
 `jsonParser = codec.getFactory().createParser(in);`

Reply:
Thanks for noting the previous approach. The line calling `setCodec()` was unchanged in the most recent updates for NIFI-12153, but it actually should have been removed as part of the changes to incorporate the constraint settings. So in this case, it is an opportunity to correct the implementation.